### PR TITLE
Support for building the extension with statically linked libzmq

### PR DESCRIPTION
**Pull request by cowo78**
Support for building the extension with statically linked libzmq.
Added missing extension dependencies on win32 (libzmq dependencies,
must be explicitly set for static builds).
[TODO] libzmq version autodetection does not work with static libzmq

### CODE REVIEW

**Comment by minrk** [comment]
Makes sense, but CI builds are failing. Maybe some checks for whether static builds should be done are incorrect?

File: setup.py [hunk 1]
```diff
@@ -363,13 +367,18 @@ def init_settings_from_config(self):
         )]
         if sys.platform.startswith('win') and sys.version_info < (3, 3):
             settings['include_dirs'].insert(0, pjoin('buildutils', 'include_win32'))
-        
+
+        settings.setdefault('libraries', [])
+        # Explicitly link dependencies, not necessary if zmq is dynamic
+        if sys.platform.startswith('win'):
+            settings['libraries'].extend(('ws2_32', 'iphlpapi', 'advapi32'))
```
Comment (minrk) [inline setup.py]: If this is only needed when libzmq is static, is it intentional that the linking unconditional?

Comment (cowo78) [reply setup.py]: I just prefer it to be explicit. Easy to change, though.

Comment (minrk) [reply setup.py]: ok, thanks

File: setup.py [hunk 2]
```diff
@@ -603,6 +612,11 @@ def bundle_libzmq_extension(self):
                 else:
                     info("ok")
 
+                if pypy:
+                    # seem to need explicit libstdc++ on linux + pypy
+                    # not sure why
+                    libzmq.libraries.append("stdc++")
```
Comment (daa) [inline setup.py]: Excuse me for disturbing you but in #1161 I intentionally disabled adding ``libstdc++`` to explicitly linked libraries under PyPy. Do you have any reasons to enable it again? And if you do please tell me, may be I'm missing something.

Comment (cowo78) [reply setup.py]: Uhmmm no. Likely my bad while merging.

**Pull request merged**